core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -87,6 +98,18 @@ public static MetricsConfig fromProperties(Map<String, String> props) {
       spec.defaultMode = MetricsModes.fromString(DEFAULT_WRITE_METRICS_MODE_DEFAULT);
     }
 
+    // First set sorted column with sorted column default (can be overridden by user)
+    final Set<String> sortedCols = new HashSet<>();
+    MetricsMode sortedColDefaultMode = promoteSortedColumnDefault(spec.defaultMode);
+    if (order != null) {
+      sortedCols.addAll(SortOrderUtil.getSortedColumns(order));
+      sortedCols.stream().forEach(sc -> {
+        if (!props.containsKey(METRICS_MODE_COLUMN_CONF_PREFIX + sc)) {
```

Comment:
Do we need this if condition? I think it is fine to default the mode here. If the user sets an explicit value, it will override this one.

Reply:
Done